### PR TITLE
[SYCL] Optimize build

### PR DESCRIPTION
**Pull request by vladimirlaz**
- Remove unused tools from the default build.
 - Use sycl target to build SYCL-related binaries.
 - Move dependencies (OpenCL ICD loader and headers)
   downloading to CMake script and do that only if no
   correspodning libraries are set in cmake command
   line or found in the system.

Signed-off-by: Vladimir Lazarev <vladimir.lazarev@intel.com>

### CODE REVIEW

**Comment by keryell** [comment]
That looks good to me.
Since it changes a lot of the build system, @agozillon can you look at how it works with https://github.com/triSYCL/sycl ?

**Comment by agozillon** [comment]
LGTM, will have to tweak it a bit when we merge again but it seems good from what I can see, I doubt it will have any negative impact. 

Might be worth updating the GetStartedWithSYCLCompiler.md at some point as the build command looks like its changed a bit.

**Comment by bader** [comment]
NOTE, the primary use for this scripts is automated validation.

Updating GetStartedWithSYCLCompiler.md is good idea, I guess CMake changes from this patch make some configuration steps optional, but existing instructions are still valid.

**Comment by agozillon** [comment]
I didn't think the changes were going to cause any issues, we just build a script to bin using the CMake infrastructure of the sycl-toolchain so worth a small test, I suppose. Works great, I had no problems integrating it.

And yeah you can still use the same configuration command from GetStartedWithSYCLCompiler.md. Although, if you use the configuration command from there with an OPENCL_HEADERS environment variable pointing at the CL directory as described in the document you seem to end up with a compile error. As the CMake build will append an extra CL directory to it. Not really a problem, just thought I'd mention it for the next iteration of the document.

Either way the changes LGTM, sorry for delaying the merge.

File: sycl/CMakeLists.txt [hunk 1]
```diff
@@ -27,12 +27,59 @@ endif()
 # Unlike PACKAGE_VERSION, CLANG_VERSION does not include LLVM_VERSION_SUFFIX.
 set(CLANG_VERSION "${CLANG_VERSION_MAJOR}.${CLANG_VERSION_MINOR}.${CLANG_VERSION_PATCHLEVEL}")
 
-set ( LLVM_INST_INC_DIRECTORY "lib${LLVM_LIBDIR_SUFFIX}/clang/${CLANG_VERSION}/include" )
+set(LLVM_INST_INC_DIRECTORY "lib${LLVM_LIBDIR_SUFFIX}/clang/${CLANG_VERSION}/include")
+set(dst_dir ${LLVM_LIBRARY_OUTPUT_INTDIR}/clang/${CLANG_VERSION}/include/CL)
```
Comment (bader) [inline sycl/CMakeLists.txt]: Please, remove "/CL".

**Comment by vladimirlaz** [comment]
> @vladimirlaz, please, update documentation.

fixed CL issue in documentation

**Pull request merged**